### PR TITLE
Features Support

### PR DESCRIPTION
**Pull request by curly210102**
When I practice with type-challenges, I notice that I need a complete tool to get through the process quickly. Then I found this plugin and made some optimizations for it.

**Effect**
31db67de83c2de209755814f2d892326846df0e1d03cf94a42e69228b1a181a0

**Feature Supported**
- [x] Pick question in challenge list
- [x] Switch languages
- [x] Copy and share answer to github issue
- [x] Quick start next question
- [x] Quick find solutions

⚠️ type-challenges support is required, and I've create a [PR](https://github.com/type-challenges/type-challenges/pull/4939).

### CODE REVIEW

**Comment by antfu** [comment]
@g-plane it would be great if you can have a quick review of it, or I think we could merge it first

File: README.md [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
-## Running this plugin
+![picture 1](images/31db67de83c2de209755814f2d892326846df0e1d03cf94a42e69228b1a181a0.gif)
 
-- [Click this link](https://www.staging-typescript.org/play?install-plugin=@type-challenges/playground-plugin) to install ([currently broken](https://github.com/microsoft/TypeScript-Website/issues/1313))
+Supported Features:
 
-or
+- [x] Pick question in challenge list
+- [x] Switch languages
+- [x] Copy and share answer to github issue
```
Comment (g-plane) [inline README.md:13]:
```suggestion
- [x] Copy and share answer to GitHub issue
```

File: README.md [hunk 2]
```diff
@@ -1,31 +1,45 @@
...
 
 ```sh
 git clone ...
-yarn install
+yarn
 yarn start
 ```
 
-Then tick the box for starting plugin development inside the TypeScript Playground.
+tick the **Connect to localhost:5000** in bottom of Plugins Panel, Refresh browser page.
```
Comment (g-plane) [inline README.md:45]:
```suggestion
tick the **Connect to localhost:5000** in bottom of Plugins Panel, and refresh browser page.
```

File: src/components/Challenges.svelte [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      e.preventDefault();
+      const url = anchorElement.getAttribute("href") ?? "";
+      const absoluteUrl = getAbsoluteLink(url);
+
+      if (absoluteUrl) {
+        window.open(absoluteUrl);
+        return;
+      }
+
+      // 切换语言
```
Comment (g-plane) [inline src/components/Challenges.svelte:59]: Using English comments will be better.

File: src/components/Challenges.svelte [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      const localeMatch = url.match(/^\.\/README(?:\.([\w-]+))?.md$/);
+      if (localeMatch) {
+        locale.set(localeMatch[1] ?? "");
+        return;
+      }
+      // 选择题目
```
Comment (g-plane) [inline src/components/Challenges.svelte:65]: Same as above. Using English comments will be better.

File: src/components/Description.svelte [hunk 5]
```diff
@@ -1,21 +1,46 @@
...
-  let markers: editor.IMarker[] = []
+  let root: HTMLElement;
+  let markers: editor.IMarker[] = [];
+  let playgroundUrl: string = "";
+  let frame: HTMLIFrameElement;
+  let loadPromise: Promise<string>;
+  let isMarkerLoaded = false;
+  $: isPassed = isMarkerLoaded && markers.length === 0;
 
   onMount(async () => {
```
Comment (g-plane) [inline src/components/Description.svelte:22]:
```suggestion
  onMount(() => {
```

File: src/components/Description.svelte [hunk 6]
```diff
@@ -1,21 +1,46 @@
...
+  async function onSwitchQuestion(quiz: number, locale: string) {
+    if (!isNaN(quiz)) {
+      const { readme, playUrl } = await fetchQuestion(quiz, locale);
+      if (playUrl) {
+        isMarkerLoaded = false;
+        playgroundUrl = playUrl;
+        setTimeout(() => {
+          isMarkerLoaded = true;
+        }, 2000);
+        return await markdownToHtml(readme, context.sandbox);
```
Comment (g-plane) [inline src/components/Description.svelte:39]: We aren't in `try`...`catch` block, so we can simplify like this:

```suggestion
        return markdownToHtml(readme, context.sandbox);
```

File: src/components/Description.svelte [hunk 7]
```diff
@@ -25,10 +50,131 @@
...
+      .getText()
+      .match(
+        /(?:\/\*\s_+\sここにコードを記入\s_+\s\*\/(.+)\/\*\s_+\sテストケース\s_+\s\*\/)|(?:\/\*\s_+\s你的代码\s_+\s\*\/(.+)\/\*\s_+\s测试用例\s_+\s\*\/)|(?:\/\*\s_+\sYour Code Here\s_+\s\*\/(.+)\/\*\s_+\sTest Cases\s_+\s\*\/)/s
+      );
+
+    const code = codeMatch ? codeMatch[1] ?? codeMatch[2] ?? codeMatch[3] : "";
+    const text = code.trim();
+    navigator.clipboard
+      .writeText(text)
+      .catch(() => {
```
Comment (g-plane) [inline src/components/Description.svelte:101]: Consider using `async`/`await` here?

**Review by g-plane — CHANGES_REQUESTED** [review]
Though I haven't run it locally, code looks OK, but there're some trivial problems.

File: src/components/Description.svelte (hunk 6 above)
Comment (curly210102) [reply src/components/Description.svelte:39]: There need to wait for code colorize to complete

https://github.com/type-challenges/playground-plugin/blob/b8dc614740784ae956523a30a744ccf4a8a6ea10/src/markdown.ts#L30-L41

Comment (g-plane) [reply src/components/Description.svelte:39]: I know, but here it just returns the result of `markdownToHtml`, so nothing different. Anyway, this isn't blocker.

**Review by g-plane — APPROVED** [review]
Thanks!

**Pull request merged**